pyccel/codegen/printing/fcode.py

```diff
@@ -1146,9 +1151,8 @@ def _print_Assign(self, expr):
 
         code = ''
         if (expr.status == 'unallocated') and not (expr.like is None):
-            stmt = ZerosLike(lhs=lhs_code, rhs=expr.like)
-            code += self._print(stmt)
-            code += '\n'
+            errors.report("Numpy array dtype=object not supported ", symbol=expr,
+                    severity='fatal')
```

Comment:
Can you explain this change?

Reply:
We discussed this. We thought that given the if condition in the line above, this case was designed to handle:
```
np.zeros_like(None)
Out[3]: array(0, dtype=object)
```
Actually looking at the line again, it seems that I misread it though. There is a `not` in the condition:
```
if (expr.status == 'unallocated') and not (expr.like is None):
```

Reply:
The line has been wrong since you updated the constructor of ZerosLike but our tests never enter this line (probably because `status` is not reliably set). It raises a codacy error. But I don't know what it's supposed to do

Reply:
If we don't know what these lines were supposed to do, I suggest we revert this changes and add a comment:
```python
# TODO [YG, 30.10.2020]: verify what this check does and removed it if obsolete
```

What kind of Codacy error do we get here?

Reply:
The codacy error is about the arguments. The constructor of `ZerosLike` is:
```
def __new__(cls, a, dtype=None, order='K', subok=True)
```
If we are lucky then Codacy will realise that this isn't actually a change (as with the examples in ast/parallel/mpi.py) but if we aren't then this will keep raising an error and will block the merge

Reply:
Unexpected keyword argument 'rhs' in constructor call
`stmt = ZerosLike(lhs=lhs_code, rhs=expr.like)`

Reply:
What if we simply comment those lines out and add a TODO comment?

Reply:
I agree with you, those lines will help in future implementation.
but I think we should also keep raising an error if so, to help in locating the problem.

Reply:
I see your point, but the error that you are raising is not correct, because there is a `not` in this condition:
```python
if (expr.status == 'unallocated') and not (expr.like is None):
```
So I'd rather just comment out the code to make Codacy happy for now.